metaseq/trainer.py

```diff
@@ -1035,6 +1035,8 @@ def _prepare_sample(self, sample, is_dummy=False):
 
         def apply_half(t):
```

Comment:
This function seems surprising now, the return value is only partly config driven

| config value | Input Value | Return Value | |
| ------ | ----- | -----| --- |
| bf16 | fp32 | bf16 | inconsistent with function name |
| bf16 | half | half | **inconsistent with config** |


I think the naming issue is less concerning, (naming is hard) but in the case where the config is bf16, and the input value is half, I think it is less surprising if we return bf16.

Reply:
does this seem better now?